### PR TITLE
WIP: Add guidance for banner size

### LINKED ISSUE #169: Anthologies: add guidance for banner dimensions

### PR DESCRIPTION
**Pull request by timdunn22**
### What does this PR do?

Makes it easy for users to add image banners to Anthologies which are of the correct type and size.

Closes #169

### For testing
1. Go to http://cove-staging.herokuapp.com/anthologies/literature-201-section-b
2. See if the whitespace for the banner looks good
3. Go to http://cove-staging.herokuapp.com/anthologies/literature-201-section-b/edit and see if the guidance looks good under the banner upload
4. try to upload an image that is jpg under  1 MB (on the banner upload part) and see if everything is working
5. try to upload an image that is jpg more than  1 MB (on the banner upload part) and see if an appropriate validation error comes up
6. try to upload a .doc file and see if a content file type error comes up

### CODE REVIEW

File: app/views/anthologies/_form.html.erb [hunk 1]
```diff
@@ -13,7 +13,7 @@
         <div class="form-group">
           <%= f.label :banner %>
           <%= f.file_field :banner, class: 'form-control' %>
-        <i>The banner will be 3000 x 420 pixels </i>
+        <i>The banner should be a png, jpg, or jpeg and should be less than 1 MB</i>
```
Comment (jamiefolsom) [inline app/views/anthologies/_form.html.erb:16]: OK, and please add the new recommended dimensions.

File: app/models/anthology.rb [hunk 2]
```diff
@@ -3,11 +3,10 @@ class Anthology < ActiveRecord::Base
   has_and_belongs_to_many :documents
   has_attached_file :banner,
                      styles: {
-    medium: ["3000x420>", :jpg],
-    small: ["1000x96>", :jpg],
-    thumb: ["800x45#", :jpg]
+    medium: ["1100x250>", :jpg],
```
Comment (jamiefolsom) [inline app/models/anthology.rb:6]: How does this deal with larger images of different aspect ratios? How does it deal with smaller images?

File: app/views/anthologies/show.html.erb [hunk 3]
```diff
@@ -17,6 +17,7 @@
       <% end %>
       <div class="media-left">
         <% unless @anthology.banner.blank? %>
+
           <%= link_to image_tag(@anthology.banner.url(:medium), class: 'media-object'), @anthology.banner.url(:medium), target: '_blank' %>
```
Comment (jamiefolsom) [inline app/views/anthologies/show.html.erb:21]: What happens if the uploaded image is smaller than the dimensions of the `:medium` version? Is a medium version created?

**Comment by timdunn22** [comment]
Closes #208 

** For testing
#1 Go to http://cove-staging.herokuapp.com/anthologies/literature-201-section-b
#2 See if the whitespace for the banner looks good
#3 Go to http://cove-staging.herokuapp.com/anthologies/literature-201-section-b/edit and see if the guidance looks good under the banner upload
#4 try to upload an image that is jpg under  1 MB (on the banner upload part) and see if everything is working
#5 try to upload an image that is jpg more than  1 MB (on the banner upload part) and see if an appropriate validation error comes up
#6 try to upload a .doc file and see if a content file type error comes up

File: app/views/anthologies/_form.html.erb (hunk 1 above)
Comment (jamiefolsom) [reply app/views/anthologies/_form.html.erb:16]: How about: "Your banner image should be a png, a gif or a jpg, 1100 pixels (or less) in width, and smaller than 1MB".

**Comment by SteveMarvin** [comment]
OK, a few issues here:

1. I need a little more guidance on what "looks good". Does this look good? Left justified OK? Less whitespace before the banner than after it? 
image
2. If the filesize is too big, the page just reloads without change. The user doesn't know whether or what they did wrong.
3. Same thing with filetype. If it doesn't take it, no error message, just nothing happens and page reload.
4. Same for taller than wide. If it doesn't take it, no error message, just nothing happens and page reload. This one isn't even written out for the user. 
5. However, it is doing what it is supposed to do, just not very informatively. Is that cool? If so, this issue can be accepted.
5.

**Comment by jamiefolsom** [comment]
@timdunn22 we need to address the fact that the validations are not giving users any feedback -- there should be an alert in the view, which shows user-friendly messages any failed validation.

**Comment by timdunn22** [comment]
# Testing
1. Login
2. Check http://cove-staging.herokuapp.com/anthologies/fourth-sample and make sure the padding from the description to the banner is sufficient (enough space between the two)
3. Go to http://cove-staging.herokuapp.com/anthologies and click on the edit link for any anthology and try to upload a non image file (.txt, .doc, etc..) that is more than 1 MB you should receive an error message at the top that indiciates the file is too large and is not of the right type.
4. Go to http://cove-staging.herokuapp.com/anthologies and click on the edit link for any anthology and try to upload an image file (should be .png, .jpg. .jpeg) that is less than 1 MB and you should be redirected to the anthologies page where you will see a message indicating the image was uploaded successfully.

File: app/models/anthology.rb (hunk 2 above)
Comment (jamiefolsom) [reply app/models/anthology.rb:6]: - [ ] Wider than 1100px; scale down proportionally to 1100 by whatever. eg 
  - 2200 x 500 scales down by 50% 
  - 2200 by 2200 >> 1100 x 1100
  - 2200 by 500 >> 1100 x 250

- [ ] Narrower than 1100px?
  - Do nothing

https://github.com/thoughtbot/paperclip/wiki/Thumbnail-Generation#resizing-options

**Comment by SteveMarvin** [comment]
Test plan passed!

**Pull request merged**